# Week7 - Decision Tree Lab

* Train-test split
* Train a decison tree model
* Train a random forest model
* Evaluate the models
* Explain findings

In [20]:

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, confusion_matrix


In [21]:

# Load the dataset
df = pd.read_csv('https://raw.githubusercontent.com/msaricaumbc/DS_data/master/ds602/log_reg/employee-turnover-balanced.csv')


In [11]:
# Split the dataset into features and target
y = df['left_company']
X = df.iloc[:, 1:]

In [3]:

# Split the dataset into a training set and a testing set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)


In [ ]:
from sklearn.preprocessing import OneHotEncoder

# one-hot encode categorical variables
encoder = OneHotEncoder()
X_encoded = encoder.fit_transform(X)

# split encoded data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X_encoded, y, test_size=0.2, random_state=42)

# train decision tree model on encoded data
dt = DecisionTreeClassifier(random_state=42)
dt.fit(X_train, y_train)

# train random forest model on encoded data
rf = RandomForestClassifier(random_state=42)
rf.fit(X_train, y_train)

# evaluate models on testing data
dt_pred = dt.predict(X_test)
rf_pred = rf.predict(X_test)

dt_acc = accuracy_score(y_test, dt_pred)
rf_acc = accuracy_score(y_test, rf_pred)

print("Decision tree accuracy:", dt_acc)
print("Random forest accuracy:", rf_acc)


In [13]:
from sklearn.preprocessing import OneHotEncoder

# one-hot encode categorical variables
encoder = OneHotEncoder()
X_encoded = encoder.fit_transform(X)

In [14]:
# split encoded data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X_encoded, y, test_size=0.2, random_state=42)


In [15]:
# train decision tree model on encoded data
dt = DecisionTreeClassifier(random_state=42)
dt.fit(X_train, y_train)


DecisionTreeClassifier(random_state=42)

In [16]:
# train random forest model on encoded data
rf = RandomForestClassifier(random_state=42)
rf.fit(X_train, y_train)


RandomForestClassifier(random_state=42)

In [17]:
# evaluate models on testing data
dt_pred = dt.predict(X_test)
rf_pred = rf.predict(X_test)


In [18]:
dt_acc = accuracy_score(y_test, dt_pred)
rf_acc = accuracy_score(y_test, rf_pred)


In [19]:
print("Decision tree accuracy:", dt_acc)
print("Random forest accuracy:", rf_acc)


Decision tree accuracy: 0.83
Random forest accuracy: 0.88


It appears that the random forest model outperformed the decision tree model on the given problem based on the accuracy scores. The accuracy score of the random forest model was 0.88, which is better than the accuracy score of the decision tree model (0.83).